### DISCUSSION TITLE
Reliable retrieval of the default Admin user password after a password change

### DISCUSSION THREAD

**Discussion by aomarai** [goharbor/harbor#19129] [Q&A]
Hi guys! I did a quick search and didn't find any recent answers as far as I could tell so wasn't sure they still applied, so here I am:

We have the Harbor containers running via a docker compose file and have full control over the machine: the operating system, root access, etc. We're trying to find a way to reliably retrieve the current password of the default Admin user so we can use it for our automation $ API purposes. We were originally looking at the environment variable for the admin password inside the harbor-core container but learned that it isn't updated after setup is completed.

I've seen that there's the harbor_user table in the internal psql db inside the harbor-db container, but the passwords seem salted + hashed. What would be the best way to reliably retrieve the current admin user's password (whether it's been previously changed or not), if any? The DB? 

Thanks!

**Comment by chlins** [comment]
The admin password by default in the ENV is only used to install harbor and initialize the default password which is not secure, when the admin user changes the password, harbor cannot retrieve it and sync it to any place which is insecure and not a best practice, so recommends storing your admin password to somewhere secure enough and fetch it by yourself.(The password is encrypted in the db you cannot reverse it to plain text.)
